pyface/tests/test_color.py

```diff
@@ -0,0 +1,246 @@
+
+from traits.testing.unittest_tools import UnittestTools
+
+from pyface.color import Color, channels_to_ints, ints_to_channels
+
+
+class TestChannelConversion(TestCase):
+
+    def test_ints_to_channels(self):
+        values = (102, 102, 0, 255)
+        channels = ints_to_channels(values)
+        self.assertEqual(channels, (0.4, 0.4, 0.0, 1.0))
+
+    def test_ints_to_channels_maximum(self):
+        values = (6, 6, 0, 15)
+        channels = ints_to_channels(values, maximum=15)
+        self.assertEqual(channels, (0.4, 0.4, 0.0, 1.0))
+
+    def test_channels_to_ints(self):
+        channels = (0.4, 0.4, 0.0, 1.0)
+        values = channels_to_ints(channels)
+        self.assertEqual(values, (102, 102, 0, 255))
+
+    def test_channels_to_ints_maximum(self):
+        channels = (0.4, 0.4, 0.0, 1.0)
+        values = channels_to_ints(channels, maximum=15)
+        self.assertEqual(values, (6, 6, 0, 15))
+
+    def test_round_trip(self):
+        """ Test to assert stability of values through round-trips """
+        for value in range(256):
+            with self.subTest(int=value):
+                result = channels_to_ints(ints_to_channels([value]))
+                self.assertEqual(result, (value,))
+
+    def test_round_trip_maximum(self):
+        """ Test to assert stability of values through round-trips """
+        for value in range(65536):
+            with self.subTest(int=value):
+                result = channels_to_ints(
+                    ints_to_channels(
+                        [value],
+                        maximum=65535,
+                    ),
+                    maximum=65535,
+                )
+                self.assertEqual(result, (value,))
+
+
+
+class TestColor(UnittestTools, TestCase):
+
+    def test_init(self):
+        color = Color()
+        self.assertEqual(color.rgba, (1.0, 1.0, 1.0, 1.0))
+
+    def test_init_rgba(self):
+        color = Color(rgba=(0.4, 0.2, 0.6, 0.8))
+        self.assertEqual(color.rgba, (0.4, 0.2, 0.6, 0.8))
+
+    def test_init_rgb(self):
+        color = Color(rgb=(0.4, 0.2, 0.6))
+        self.assertEqual(color.rgba, (0.4, 0.2, 0.6, 1.0))
+
+    def test_init_r_g_b_a(self):
+        color = Color(red=0.4, green=0.2, blue=0.6, alpha=0.8)
+        self.assertEqual(color.rgba, (0.4, 0.2, 0.6, 0.8))
+
+    def test_init_r_g_b(self):
+        color = Color(red=0.4, green=0.2, blue=0.6)
+        self.assertEqual(color.rgba, (0.4, 0.2, 0.6, 1.0))
+
+    def test_init_hsva(self):
+        color = Color(hsva=(0.4, 0.2, 0.6, 0.8))
+        self.assertEqual(color.rgba, (0.48, 0.6, 0.528, 0.8))
+
+    def test_init_hsv(self):
+        color = Color(hsv=(0.4, 0.2, 0.6))
+        self.assertEqual(color.rgba, (0.48, 0.6, 0.528, 1.0))
+
+    def test_init_hlsa(self):
+        color = Color(hlsa=(0.4, 0.2, 0.6, 0.8))
+        self.assertEqual(
+            color.rgba,
+            (0.07999999999999996, 0.32000000000000006, 0.17600000000000007, 0.8)  # noqa: E501
+        )
```

Comment:
nit : it might be cleaner if we can use `assertAlmostEqual` and set a tolerance?

Reply:
`AssertAlmostEqual` doesn't support tuples of floats, only floats themselves.

I'll write a method to do the almost-equals comparison.